UPGRADING.md

```diff
@@ -0,0 +1,183 @@
+# Microsoft Graph PHP SDK Upgrade Guide
+
+This guide highlights backward compatibility breaking changes introduced during major upgrades.
+
+
+## 1.x to 2.0
+
+Version `2.0` highlights:
+- Support for National Clouds.
+- Changes in creating a Graph client.
+- Changes in configuring your HTTP client (including support for PSR-18 and HTTPlug's HttpAsyncClient implementations).
+- Introducing standardised Graph exception types `GraphClientException` and `GraphServiceException` as more specific `GraphException` types.
```

Comment:
It would be helpful to link documentation for these changes

Reply:
Added